Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -106,37 +122,73 @@ void visitSeedIdentifiers(const TrackProxy& track, Visitor visitor) {
   }
 }
 
-}  // namespace
-}  // namespace ActsExamples
+class BranchStopper {
+ public:
+  using Config =
+      std::optional<std::variant<Acts::TrackSelector::Config,
+                                 Acts::TrackSelector::EtaBinnedConfig>>;
 
-// Specialize std::hash for SeedIdentifier
-// This is required to use SeedIdentifier as a key in an `std::unordered_map`.
-template <class T, std::size_t N>
-struct std::hash<std::array<T, N>> {
-  std::size_t operator()(const std::array<T, N>& array) const {
-    std::hash<T> hasher;
-    std::size_t result = 0;
-    for (auto&& element : array) {
-      boost::hash_combine(result, hasher(element));
+  explicit BranchStopper(const Config& config) : m_config(config) {}
+
+  bool operator()(
+      const Acts::CombinatorialKalmanFilterTipState& tipState,
+      Acts::VectorMultiTrajectory::TrackStateProxy& trackState) const {
+    if (!m_config.has_value()) {
+      return false;
     }
-    return result;
+
+    bool haveCuts = std::visit(
+        [&](const auto& config) -> bool {
+          using T = std::decay_t<decltype(config)>;
+          if constexpr (std::is_same_v<T, Acts::TrackSelector::Config>) {
+            return true;
+          } else if constexpr (std::is_same_v<
+                                   T, Acts::TrackSelector::EtaBinnedConfig>) {
+            double theta = trackState.parameters()[Acts::eBoundTheta];
+            double eta = -std::log(std::tan(0.5 * theta));
+            return config.hasCuts(eta);
+          }
+        },
+        *m_config);
+
+    if (!haveCuts) {
+      return false;
+    }
+
+    const Acts::TrackSelector::Config& singleConfig = std::visit(
+        [&](const auto& config) -> const Acts::TrackSelector::Config& {
+          using T = std::decay_t<decltype(config)>;
+          if constexpr (std::is_same_v<T, Acts::TrackSelector::Config>) {
+            return config;
+          } else if constexpr (std::is_same_v<
+                                   T, Acts::TrackSelector::EtaBinnedConfig>) {
+            double theta = trackState.parameters()[Acts::eBoundTheta];
+            double eta = -std::log(std::tan(0.5 * theta));
+            return config.getCuts(eta);
+          }
+        },
+        *m_config);
+
+    if (tipState.nMeasurements >= singleConfig.minMeasurements) {
```

Comment:
```suggestion
    if (tipState.nMeasurements >= singleConfig.minMeasurements &&
        !trackState.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
```
Prevent the branch being stopped (and therefore discarded) if there are missing measurements at the end of an otherwise perfectly good track.